src/com/gr1t/taskmanager/TaskController.java

```diff
@@ -0,0 +1,186 @@
+package com.gr1t.taskmanager;
+
+import java.util.List;
+import java.util.Scanner;
+
+public class TaskController {
+    private static final Scanner scanner = new Scanner(System.in);
+
+    public static void start() {
+        while (true) {
+            showMenu();
+            int choice = getUserChoice();
+
+            switch (choice){
+                case 1:
+                    createTask();
+                    break;
+                case 2:
+                    viewTasks();
+                    break;
+                case 3:
+                    saveAsFile();
+                    break;
+                case 4:
+                    sortByPriority();
+                    break;
+                case 5:
+                    sortByStatus();
+                    break;
+                case 6:
+                    markCompleted();
+                    break;
+                case 7:
+                    removeByNumber();
+                    break;
+                case 8:
+                    updateTask();
+                    break;
+                case 9:
+                    return;
+                default:
+                    System.out.println("Value is not correct, please try again");
+            }
+
+        }
+    }
+    
+    private static void showMenu(){
+        System.out.println("1. Create Task");
+        System.out.println("2. View All Tasks");
+        System.out.println("3. Save as file");
+        System.out.println("4. Sort by priority");
+        System.out.println("5. Sort by status");
+        System.out.println("6. Make task completed");
+        System.out.println("7. Remove by number");
+        System.out.println("8. Update task");
+        System.out.println("9. Exit");
+        System.out.print("Choose an option: ");
+    }
+
+    private static int getUserChoice() {
+        int choice;
+        while (true) { // Check input for number
+            if (!scanner.hasNextInt()) {
+                System.out.println("Invalid input. Enter a number:");
+                scanner.next();
+                continue;
+            }
+
+            choice = scanner.nextInt();
+
+            if (choice < 1 || choice > 9) { // Check input for correct range
+                System.out.println("Invalid input. Please enter a number in range 1-3: ");
+                continue;
+            }
+            break;
+        }
+        return choice;
+    }
+
+    private static void createTask() {
```

Comment:
Controller shouldn't be responsible for the Task CRUD

Reply:
Pan rainsxngreview, do you mean I should change method's names?